grails-app/services/org/pih/warehouse/receiving/ReceiptService.groovy

```diff
@@ -287,26 +294,39 @@ class ReceiptService {
         }
     }
 
+    /**
+     We can create RECEIVED or PARTIALLY_RECEIVED event depending on the case:
+        1. When the location supports partial receiving
+            RECEIVED - when the shipment is not already received and is fully received
+            PARTIALLY_RECEIVED - when the shipment wasn't partially received before (it's
+                created only once)
+        2. When the location doesn't support partial receiving
+            RECEIVED - after receiving for the first time, we cannot receive
+                for the second time without partial receiving. The rest of the remaining quantities
+                should be canceled
+            PARTIALLY_RECEIVED - not allowed
+     */
     void savePartialReceiptEvent(PartialReceipt partialReceipt) {
         Shipment shipment = partialReceipt.shipment
         Receipt receipt = partialReceipt.receipt
 
-        if (shipment.isFullyReceived()) {
-            if (!shipment.wasReceived()) {
-                shipmentService.createShipmentEvent(shipment,
-                        receipt?.actualDeliveryDate,
-                        EventCode.RECEIVED,
-                        shipment.destination)
-            }
-        } else {
+        if (!shipment.wasReceived() && (!shipment.destination.supports(ActivityCode.PARTIAL_RECEIVING) || shipment.isFullyReceived())) {
```

Comment:
why are we even allowing the flow to continue here if partial receiving is disabled for the location? Shouldn't we error instead?

Based on the fact that we don't error I'm guessing it's because regular receiving flows also use this method, which is misleading IMO based on the method name if that's true. Anyways, not new code so nothing I expect you to change.

Reply:
I don't know why we call this feature "partial" everywhere, because all those functions simultaneously support partial / not partial receiving. That's the reason for allowing it here; if partial receiving is disabled, we can receive it only once.